### ISSUE TITLE
[Formsy.Form API] reset() should accept empty values

### ISSUE THREAD

**Issue by Grmiade** [christianalfoni/formsy-react#314]
I have a form initialized like this :

```javascript
<FormsyText
  name="email"
  value={this.props.email}
  type="email"
  validations="isEmail"
  validationError={this.context.i18n.t('You have to type a valid email.')}
/>
```

If `this.props.email` is not empty, this value is displayed. But next when i reset the form with a empty value like this:

```javascript
nextProps = {
  phone: "+336574xxxx"
  email: "",
}
this.refs.form.reset(nextProps);
```

The value of my field not changed. Indeed "To pass an initial value, use the value attribute. This value will become the "pristine" value and any reset of the form will bring back this value."
But an explicit empty value should force this behavior, no ? How I can reset my field with a empty value after a not empty value ?

**Comment by Grmiade** [comment]
I think that the PR #259 could help me. Any news ?

**Comment by Semigradsky** [comment]
Fixed in 6767a6b9eba9c0a7c255c46ce2d6f84dc69eeeb4
